src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -444,6 +452,57 @@ private void SetTransferMode(TransferMode transferMode)
             _httpClient.Dispose();
         }
 
+        public IDisposable OnReceived(Func<byte[], object, Task> callback, object state)
+        {
+            var callBack = new ReceiveCallBack(callback, state);
+            lock (_callBacks)
+            {
+                _callBacks.Add(callBack);
+            }
+            return new Subscription(callBack, _callBacks);
+        }
+
+        public IDisposable OnReceived(Func<byte[], Task> callback)
+        {
+            return OnReceived((data, _) => { return callback(data); }, null);
+        }
+
+        public class ReceiveCallBack
```

Comment:
private

Reply:
So I'm also using these classes in TestConnection. I'll just copy them then